Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/connectmgr/ConnectionManagerCache.java

```diff
@@ -866,17 +869,25 @@ public String getEndpointURLFromNhinTarget(NhinTargetSystemType targetSystem, St
             } else if (targetSystem.getHomeCommunity() != null
                 && NullChecker.isNotNullish(targetSystem.getHomeCommunity().getHomeCommunityId())
                 && NullChecker.isNotNullish(serviceName)) {
-
                 // Get the URL based on Home Community Id and Service Name
                 String homeCommunityId = HomeCommunityMap.formatHomeCommunityId(targetSystem.getHomeCommunity()
                     .getHomeCommunityId());
-                LOG.debug("Attempting to look up URL by home communinity id: " + homeCommunityId
-                    + " and service name: " + serviceName);
-                sEndpointURL = getDefaultEndpointURLByServiceName(homeCommunityId, serviceName);
+                final String userSpecVersion = targetSystem.getUseSpecVersion();
```

Comment:
if variable is not modifying anywhere else, it is good practice to declare them as final to prevent unseen assignment.